external_parser/event_processors/joined_event.h

```diff
@@ -300,6 +300,109 @@ struct ccb_joined_event : public typed_joined_event {
   }
 };
 
+// MOVE TO VW
+struct DecisionServiceInteraction_CA
+{
+  std::string eventId;
+  std::string timestamp;
+  unsigned action;
+  float pdf_value;
+  float probabilityOfDrop = 0.f;
+  bool skipLearn{false};
+};
+
+struct ca_joined_event : public typed_joined_event {
+  DecisionServiceInteraction_CA interaction_data;
+  // Default Baseline Action for CB is 1 (rl client recommended actions are 1
+  // indexed in the CB case)
+  float reward = 0.0f;
+  float original_reward = 0.0f;
+
+  ~ca_joined_event() = default;
+
+  bool is_skip_learn() const override { return interaction_data.skipLearn; }
+
+  void set_skip_learn(bool sl) override { interaction_data.skipLearn = sl; }
+
+  // TODO: Add learning_mode to CaEvent
+  void set_apprentice_reward() override {
+    /*if (!interaction_data.actions.empty() &&
```

Comment:
nit: remove commented out code and just have an empty function with a comment saying that apprentice isn't implemented for CA yet

Reply:
Added learning_mode so this doesnt need to be empty anymore